src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,63 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private LinkedList<Entry<K, V>>[] buckets;
+    private int size;
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public MyHashMap() {
+        buckets = new LinkedList[INITIAL_CAPACITY];
+        size = 0;
+    }
+
+    private int getBucketIndex(K key) {
+        return Math.abs(key.hashCode()) % buckets.length;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getBucketIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new LinkedList<>();
+        }
+        for (Entry<K, V> entry: buckets[index]) {
+            if (entry.key.equals(key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        buckets[index].add(new Entry<>(key, value));
+        size++;
     }
```

Comment:
The `put` method does not handle resizing when the load factor is exceeded. Implementing a resize mechanism is crucial for maintaining performance as more elements are added.